src/librustc_codegen_llvm/callee.rs

```diff
@@ -29,7 +29,9 @@ pub fn get_fn(cx: &CodegenCx<'ll, 'tcx>, instance: Instance<'tcx>) -> &'ll Value
 
     assert!(!instance.substs.needs_infer());
     assert!(!instance.substs.has_escaping_bound_vars());
-    assert!(!instance.substs.has_param_types());
+    if !tcx.sess.opts.debugging_opts.polymorphize {
+        assert!(!instance.substs.has_param_types());
```

Comment:
You can just remove this, the other two asserts are the important ones.

Reply:
Fixed in 5b3c02b.